search/utils.py

```diff
@@ -65,3 +70,15 @@ def upper_string(self):
     def lower_string(self):
         """ use isoformat for _lower date's string format """
         return self._lower.isoformat()
+
+
+def doc_type_warning_decorator(func):
+    """
+    Decorator which mark function with doc_type as deprecated.
+    """
+    @functools.wraps(func)
+    def wrapper(ref, *args, **kwargs):
```

Comment:
What do you mean `ref`?

Reply:
It can be `self` or `cls` attribute

Reply:
Do it's a good idea to call `self_or_cls` in my opinion. What's the problem  🙂

Reply:
Done